From 1e596a5b6e72cfca67a3eeb7d9098d015a295545 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C5=BDidek?= <mzidek@redhat.com>
Date: Mon, 19 Oct 2015 15:38:08 +0200
Subject: [PATCH 094/108] util: Continue if setlocale fails
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

setlocale needs some environment variables
to be set in order to work. These variables
are not present in some special cases. We
should not fail completely in these cases
but continue with the compatible C locale.

Resolves:
https://fedorahosted.org/sssd/ticket/2785

Reviewed-by: Michal Židek <mzidek@redhat.com>
(cherry picked from commit 43e06ff39584570817949dc5de118d2b7ca854c1)
(cherry picked from commit 4815471669a25566f6772c228c104a206ffa37f7)
---
 src/sss_client/ssh/sss_ssh_client.c | 4 +++-
 src/tools/tools_util.c              | 4 +++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/sss_client/ssh/sss_ssh_client.c b/src/sss_client/ssh/sss_ssh_client.c
index 0d206ef58f3a1353a8e066c272df0cf4e22ec4db..a198039ec4e2820ad388be41400411753459ecc9 100644
--- a/src/sss_client/ssh/sss_ssh_client.c
+++ b/src/sss_client/ssh/sss_ssh_client.c
@@ -50,7 +50,9 @@ int set_locale(void)
 
     c = setlocale(LC_ALL, "");
     if (c == NULL) {
-        return EIO;
+        /* If setlocale fails, continue with the default
+         * locale. */
+        DEBUG(SSSDBG_MINOR_FAILURE, "Unable to set locale\n");
     }
 
     errno = 0;
diff --git a/src/tools/tools_util.c b/src/tools/tools_util.c
index 3112171a6c06a50b7099f1c1b58ea1ad581c7cb3..82462f30690e4f695477eb5cc1aa4039c05109f9 100644
--- a/src/tools/tools_util.c
+++ b/src/tools/tools_util.c
@@ -259,7 +259,9 @@ int set_locale(void)
 
     c = setlocale(LC_ALL, "");
     if (c == NULL) {
-        return EIO;
+        /* If setlocale fails, continue with the default
+         * locale. */
+        DEBUG(SSSDBG_MINOR_FAILURE, "Unable to set locale\n");
     }
 
     errno = 0;
-- 
2.7.3

